### PR TITLE
Add package to library component in package.yaml

### LINKED ISSUE #1230: Add Package command does nothing on package.yaml for library component

### PR DESCRIPTION
**Pull request by fendor**
Closes #1230.

Adds some unit-tests.

Todo: 
* [x] Code tidy up
* [x] More tests

### CODE REVIEW

File: test/utils/TestUtils.hs [hunk 1]
```diff
@@ -164,8 +168,7 @@ logFilePath = "functional-hie-" ++ stackYaml ++ ".log"
 -- We also need to unset STACK_EXE manually inside the tests if they are
 -- run with `stack test`
 hieCommand :: String
-hieCommand = "stack exec --no-stack-exe --no-ghc-package-path --stack-yaml=" ++ stackYaml ++
-             " hie -- -d -l test-logs/" ++ logFilePath
+hieCommand = "hie -d -l test-logs/" ++ logFilePath
```
Comment (bubba) [inline test/utils/TestUtils.hs]: We need to run hie via stack in order to get the right version we’re testing

Comment (fendor) [reply test/utils/TestUtils.hs]: Yes, I committed that accidentally! Will be fixed ASAP, thanks!

**Comment by fendor** [comment]
It seems that the tests fail for cabal files and ghc versions below 8.4.*. 
They probably have always failed, but the additional tests have shown their failure. 
What is the preferred course of action? Disabling the feature altogether? I could also try to fix it for the old ghc-versions.

**Comment by bubba** [comment]
@fendor did the FunctionalCodeActionsSpec tests for "add package suggestions" not used to pass on 8.2.x? I presume if #1249 is happening this won't matter, too much, but out of curiosity how were they failing?

**Comment by fendor** [comment]
The problem can acutally be shown in the Unit-tests.
Given the cabal file and task to add the package `text`:

Example .cabal: 
```haskell
name:                add-package-test
version:             0.1.0.0
license:             BSD3
author:              Luke Lau
maintainer:          luke_lau@icloud.com
build-type:          Simple
extra-source-files:  ChangeLog.md
cabal-version:       >=1.10

executable AddPackage
  exposed-modules:     ./.
  main-is:             AddPackage.hs
  build-depends:       base >=4.7 && <5
  default-language:    Haskell2010
```
Normally the result should be:
```haskell
              List
                [ TextEdit (Range (Position 0 0) (Position 7 27)) $ T.concat
                  [ "cabal-version: >=1.10\n"
                  , "name: add-package-test\n"
                  , "version: 0.1.0.0\n"
                  , "license: BSD3\n"
                  , "maintainer: luke_lau@icloud.com\n"
                  , "author: Luke Lau\n"
                  , "build-type: Simple\n"
                  , "extra-source-files:\n"
                  , "    ChangeLog.md"
                  ]
                , TextEdit (Range (Position 10 0) (Position 13 34)) $ T.concat
                  [ "    main-is: AddPackage.hs\n"
                  , "    default-language: Haskell2010\n"
                  , "    build-depends:\n"
                  , "        base >=4.7 && <5,\n"
                  , "        text -any"
                  ]
                ]
```

But on GHC 8.2.* the output is actually: 
```haskell
              List -- TODO: this seems to indicate that the command does nothing
                [ TextEdit (Range (Position 0 0) (Position 7 27)) $ T.concat
                  [ "name: add-package-test\n"
                  , "version: 0.1.0.0\n"
                  , "cabal-version: >=1.10\n"
                  , "build-type: Simple\n"
                  , "license: BSD3\n"
                  , "maintainer: luke_lau@icloud.com\n"
                  , "author: Luke Lau\n"
                  , "extra-source-files:\n"
                  , "    ChangeLog.md"
                  ]
                , TextEdit (Range (Position 9 0) (Position 13 34)) $ T.concat
                  [ "executable  AddPackage\n"
                  , "    main-is: AddPackage.hs\n"
                  ]
                ]
```
It seems like the `build-depends` field has either been swallowed, or hadnt been updated.

**Comment by fendor** [comment]
I think, we can actually merge this. It adds a bunch of useful tests and the problem this PR uncovered is present anyway. At least, it will be documented via tests now.

**Comment by alanz** [comment]
@fendor Please update this branch to work with current master, and we can merge it again.

**Pull request merged**